grant-project/grant-monorepo
# AddressInput component & Donation address

Closes #77.

### What This Does

* Adds a donation address map to `utils/constants`
* Adds an AddressInput component
    * Placeholder is donation address by default
    * Checks eth address validity
    * Optionally shows identicon in input
* Creates a storybook for AddressInput

### What This Doesn't Do

* Add donation address to the footer. That whole thing is going to need a separate pass of design love once we have all of our pages more finalized. I'll save that until then.
* Replace _all_ address entries with AddressInput. The stuff in the create flow was a little too custom for me to feel it was worth it. Open to opinions on this one.

### Steps to Test

* Go to sign in
* Choose to specify an address
* Enter a valid address, confirm identicon looks ok
* Enter an invalid address, confirm you get an error

## Screenshots

<img width="337" alt="screen shot 2018-10-19 at 5 38 17 pm" src="https://user-images.githubusercontent.com/649992/47245152-2571cb00-d3c6-11e8-87e9-182e0a1faf33.png">
